chapter03/dags/01_unscheduled.py

```diff
@@ -4,41 +4,42 @@
 from airflow import DAG
 from airflow.operators.bash import BashOperator
 from airflow.operators.python import PythonOperator
-from pendulum import datetime
+import pendulum
 
 
 def _calculate_stats(input_path, output_path):
     """Calculates event statistics."""
 
-    events = pd.read_json(input_path)
-    stats = events.groupby(["date", "user"]).size().reset_index()
+    events = pd.read_json(input_path, convert_dates=["timestamp"])
+
+    stats = (events
+             .assign(date=lambda df: df["timestamp"].dt.date)
+             .groupby(["date", "user"]).size().reset_index()
+    )
 
     Path(output_path).parent.mkdir(exist_ok=True)
     stats.to_csv(output_path, index=False)
 
 
 with DAG(
-    dag_id="L06_query_with_dates",
-    schedule="@daily",
-    start_date=datetime(2024, 1, 1),
-    end_date=datetime(2024, 1, 5),
+    dag_id="01_unscheduled",
+    schedule=None,
 ):
     fetch_events = BashOperator(
         task_id="fetch_events",
         bash_command=(
-            "curl -o /data/events.json "
-            "http://events_api:5000/events?"
-            "start_date=2024-01-01&"
-            "end_date=2024-01-02"
+            "mkdir -p /data/01_unscheduled && "
+            "curl -o /data/01_unscheduled/events.json"
+            " http://events-api:8081/events/latest"
```

Comment:
Compose says port 5000?

Reply:
Nice catch! The API container itself actually exposes 8081, so you end up with both 5000 and 8081:
```
0.0.0.0:5000->5000/tcp, 8081/tcp
```
All code was using 8081, so I've switched it over in the compose now too.